library/kafka_users.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""
+Ansible module for users configuration management
+"""
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+# Init logging
+import logging
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.kafka_lib_user import process_module_users
+
+from ansible.module_utils.kafka_lib_commons import (
+    module_commons, DOCUMENTATION_COMMON
+)
+# Default logging
+# TODO: refactor all this logging logic
+log = logging.getLogger('kafka')
+log.addHandler(logging.StreamHandler(sys.stdout))
+log.setLevel(logging.INFO)
+
+log = logging.getLogger('kazoo.client')
+log.addHandler(logging.StreamHandler(sys.stdout))
+log.setLevel(logging.INFO)
+
+ANSIBLE_METADATA = {'metadata_version': '1.0'}
+
+
+DOCUMENTATION = '''
+---
+module: kafka_users
+short_description: Manage Kafka users
+description:
+     - Configure Kafka users.
+     - Not compatible avec Kafka version < 0.11.0.
+author:
+    - Stephen SORRIAUX
+    - ryarnyah
+options:
+  users:
+    description:
+      - users to create. @See kafka_topic for options
+  
+
+  ... tbc ...
+''' + DOCUMENTATION_COMMON
+
+EXAMPLES = '''
+
+    # creates users with provided configuation for plaintext
+    - name: create users
+      kafka_users:
+        users:
+          - name: 'user1'
+            password: changeit
+            mechanism: SCRAM-SHA-512
+            state: 'present'
+          - name: 'user2'
+            mechanism: SCRAM-SHA-512
+            state: 'absent'
+        bootstrap_servers: >
+          "{{ hostvars['kafka1']['ansible_eth0']['ipv4']['address'] }}:9092,
+          {{ hostvars['kafka2']['ansible_eth0']['ipv4']['address'] }}:9092"
+
+'''
+
+
+def main():
+    """
+    Module usage
+    """
+    spec = dict(
+        users=dict(
+            type='list',
+            elements='dict',
+            required=True,
+            options=dict(
+                name=dict(type='str', required=True),
```

Comment:
This dict with user options may be added in common file to be reused in both modules.